sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/AppendFilesToTables.java

```diff
@@ -201,14 +200,15 @@ private void appendManifestFiles(Table table, Iterable<FileWriteResult> fileWrit
     }
 
     private ManifestWriter<DataFile> createManifestWriter(
-        String tableLocation, String uuid, PartitionSpec spec, FileIO io) {
+        String tableLocation, String uuid, PartitionSpec spec, Table table) {
```

Comment:
if we are passing `Table` now we could drop `tableLocation` and `spec`

Reply:
Thx for the catch. I refactored this a lil to make it more clean.

btw `spec` here isn't a simple `table.spec()`. This part of the code deals with the edge case where we have a batch of data files that were written with different specs (e.g. user updates spec during pipeline runtime). We group data files by spec and create/append one ManifestFile per spec.

Reply:
ah! nice makes sense